### PR TITLE
Add tests for indexing arrays with IdentityUnitRanges

### PR DESCRIPTION
**Pull request by jishnub**
Tests with `Base.IdentityUnitRange` were missed out in #39896, would be good to include these as well

### CODE REVIEW

File: test/abstractarray.jl [hunk 1]
```diff
@@ -1295,11 +1295,20 @@ end
 
 @testset "issue #39896, modified getindex " begin
     for arr = ([1:10;], reshape([1.0:16.0;],4,4), reshape(['a':'h';],2,2,2))
-        for inds = (2:5, Base.OneTo(5), BigInt(3):BigInt(5), UInt(4):UInt(3))
+        for inds = (2:5, Base.OneTo(5), BigInt(3):BigInt(5), UInt(4):UInt(3),
+            Base.IdentityUnitRange(Base.OneTo(4)), Base.IdentityUnitRange(2:4))
```
Comment (simeonschaub) [inline test/abstractarray.jl:1299]:
```suggestion
                    Base.IdentityUnitRange(Base.OneTo(4)), Base.IdentityUnitRange(2:4))
```

File: test/abstractarray.jl [hunk 2]
```diff
@@ -1295,11 +1295,20 @@ end
...
             @test arr[inds] == arr[collect(inds)]
             @test arr[inds] isa AbstractVector{eltype(arr)}
         end
     end
+    # Test that ranges and arrays behave identically for indices with 1-based axes
+    for r in (1:10, 1:1:10, Base.OneTo(10),
+        Base.IdentityUnitRange(Base.OneTo(10)), Base.IdentityUnitRange(1:10))
```
Comment (simeonschaub) [inline test/abstractarray.jl:1306]:
```suggestion
              Base.IdentityUnitRange(Base.OneTo(10)), Base.IdentityUnitRange(1:10))
```

**Review by simeonschaub — APPROVED** [review]
Small indentation nit, but otherwise LGTM.

**Comment by vtjnash** [comment]
Tried to merge this, but the tests failed. See commit message of 1b5466906b17a01b548f4239a3f0a41eab64a5bd

**Comment by vtjnash** [comment]
Possibly broken by #40660?

**Comment by jishnub** [comment]
Indeed, should be fixed now. Tests for `IdentityUnitRange(::UnitRange)` only work with `OffsetArrays` loaded, and were added by #40660 anyway. Test failures on windows seem unrelated

**Pull request merged**